lib/datadog/appsec/contrib/rack/gateway/watcher.rb

```diff
@@ -26,23 +28,25 @@ def self.watch
                   trace = active_trace
                   span = active_span
 
-                  Rack::Reactive::Request.subscribe(op, waf_context) do |action, result, _block|
-                    record = [:block, :monitor].include?(action)
+                  Rack::Reactive::Request.subscribe(op, waf_context) do |result, _block|
+                    record = [:match].include?(result.status)
                     if record
                       # TODO: should this hash be an Event instance instead?
                       event = {
                         waf_result: result,
                         trace: trace,
                         span: span,
                         request: request,
-                        action: action
+                        actions: result.actions
                       }
 
+                      span.set_tag('appsec.event', 'true') if span
```

Comment:
When Tracing is disabled.

In that case AppSec code is supposed to run, even though it cannot record information in the trace†.

With the current state as of this PR it may seem useless (since results essentially go to `/dev/null`), but this becomes more important as e.g in-app request blocking is being progressively introduced ([hint,hint!](https://github.com/DataDog/dd-trace-rb/pull/2316))

I tried hard for this case to keep working, as it's tricky to reintroduce after the fact.

† That may change if (when?) "traces" become some sort of generic data bus upon which components such as Tracing and AppSec are built.